src/Interpreters/ActionsDAG.cpp

```diff
@@ -219,6 +219,9 @@ const ActionsDAG::Node & ActionsDAG::addFunction(
             column = node.function_base->getConstantResultForNonConstArguments(arguments, node.result_type);
         }
 
+        if (all_const && column && !isColumnConst(*column) && column->size() <= 1)
+            column = ColumnConst::create(std::move(column), column->size());
```

Comment:
In order to be consistent, it needs to be removed here in `analyzeAggregation`:
https://github.com/ClickHouse/ClickHouse/blob/e8407dc930ee7e2c6c7891b4c91a94ab93fb707b/src/Interpreters/ExpressionAnalyzer.cpp#L252-L261
And in order to be removed - a non-null column needs to be added here:
https://github.com/ClickHouse/ClickHouse/blob/e8407dc930ee7e2c6c7891b4c91a94ab93fb707b/src/Interpreters/ActionsVisitor.cpp#L1061
It will be not null and added in this line only if it is columnConst:
https://github.com/ClickHouse/ClickHouse/blob/970af529a09082419d4eee74097c927922ec419d/src/Interpreters/ActionsDAG.cpp#L223-L233
It also works if casting to `columnConst` in case result column is size 1 in `IFunction::execute` and seems it is also ok - may be it will be better to cast there?